relayer/processor/path_end_runtime.go

```diff
@@ -493,9 +493,13 @@ func (pathEnd *pathEndRuntime) shouldSendChannelMessage(message channelIBCMessag
 			toDeleteCounterparty[chantypes.EventTypeChannelOpenAck] = []ChannelKey{counterpartyKey}
 			toDelete[chantypes.EventTypeChannelOpenTry] = []ChannelKey{channelKey}
 			toDeleteCounterparty[chantypes.EventTypeChannelOpenInit] = []ChannelKey{counterpartyKey.msgInitKey()}
+		case chantypes.EventTypeChannelCloseConfirm:
+			toDeleteCounterparty[chantypes.EventTypeChannelCloseInit] = []ChannelKey{counterpartyKey}
```

Comment:
We also need to delete all of the relevant packet flow messages in this flow for the relative channel key, e.g:
- `toDeleteCounterpartyPacket[EventTypeTimeoutPacket]`
- `toDeletePacket[chantypes.EventTypeSendPacket]`

these can be constructed by iterating over either `pathEnd` or `counterparty` `messagesCache.PacketFlow[channelKey or counterpartyKey, respectively]`

Reply:
AH okay. I misread this the first time I went through it this morning but that makes sense. Will get this adjusted in a bit here.